src/cli/javascript/request_filter.rs

```diff
@@ -34,14 +34,25 @@ impl RequestFilter {
     }
 
     #[async_recursion::async_recursion]
-    async fn on_request(&self, mut request: reqwest::Request) -> anyhow::Result<Response<Bytes>> {
+    async fn on_request(
+        &self,
+        mut request: reqwest::Request,
+        http_filter: Option<http::HttpFilter>,
+    ) -> anyhow::Result<Response<Bytes>> {
+        // by default
+        let mut on_request = "onRequest".to_string();
+
+        if let Some(value) = http_filter {
+            on_request = value.on_request;
+        }
+
         let js_request = JsRequest::try_from(&request)?;
         let event = Event::Request(js_request);
-        let command = self.worker.call("onRequest".to_string(), event).await?;
+        let command = self.worker.call(on_request, event).await?;
```

Comment:
worker should only be called, if the filter is not none.

Reply:
ok.

Reply:
also add if a check that on-request is not none.